### DISCUSSION TITLE
Infinite loop in grouping the arraysfield of objects by key

### DISCUSSION THREAD

**Discussion by leoso0611** [react-hook-form/react-hook-form#5718] [Q&A]
I am working on mapping an arraysfield to objects of arrays according to one key, in ease of fitting the frontend displaying.

`tempData: [
    {
        'type': '1',
        'label': 'label1'
    }, {
        'type': '1',
        'label': 'label2'
    }, {
        'type': '2',
        'label': 'label3'
    }, {
        'type': '2',
        'label': 'label4'
    }, {
        'type': 'kia',
        'label': 'label5'
    },
];`

to

`{
 1: [{
        'type': '1',
        'label': 'label1'
    }, {
        'type': '1',
        'label': 'label2'
    }],
 2: [{
        'type': '2',
        'label': 'label3'
    }, {
        'type': '2',
        'label': 'label4'
    ],
 3: [{
        'type': 'kia',
        'label': 'label5'
    }]
}`

As I need to the latest updated data, so I used the `watch()` for doing the grouping. But it will trigger an infinite re-rendering. 
Any solution and suggestion on this issue.

**Comment by bluebill1049** [comment]
can you provide a CSB?
